stellar/its.js

```diff
@@ -23,6 +39,156 @@ async function removeTrustedChain(wallet, _, chain, contractConfig, arg, options
     await broadcast(operation, wallet, chain, 'Trusted Chain Removed', options);
 }
 
+async function deployInterchainToken(wallet, _, chain, contractConfig, args, options) {
+    const contract = new Contract(contractConfig.address);
+    const caller = nativeToScVal(Address.fromString(wallet.publicKey()), { type: 'address' });
+    const minter = caller;
+    const [symbol, name, decimal, salt, initialSupply] = args;
+    const saltBytes32 = isHexString(salt) ? hexZeroPad(salt, 32) : keccak256(salt);
+
+    const operation = contract.call(
+        'deploy_interchain_token',
+        caller,
+        nativeToScVal(Buffer.from(arrayify(saltBytes32)), { type: 'bytes' }),
+        tokenMetadataToScVal(decimal, name, symbol),
+        nativeToScVal(initialSupply, { type: 'i128' }),
+        minter,
+    );
+
+    await broadcast(operation, wallet, chain, 'Interchain Token Deployed', options);
+}
+
+async function deployRemoteInterchainToken(wallet, _, chain, contractConfig, args, options) {
+    const contract = new Contract(contractConfig.address);
+    const caller = nativeToScVal(Address.fromString(wallet.publicKey()), { type: 'address' });
+    const [salt, destinationChain, gasTokenAddress, gasFeeAmount] = args;
+    const saltBytes32 = hexZeroPad(salt.startsWith('0x') ? salt : '0x' + salt, 32);
+
+    const operation = contract.call(
+        'deploy_remote_interchain_token',
+        caller,
+        nativeToScVal(Buffer.from(arrayify(saltBytes32)), { type: 'bytes' }),
+        nativeToScVal(destinationChain, { type: 'string' }),
+        tokenToScVal(gasTokenAddress, gasFeeAmount),
+    );
+
+    await broadcast(operation, wallet, chain, 'Remote Interchain Token Deployed', options);
+}
+
+async function registerCanonicalToken(wallet, _, chain, contractConfig, args, options) {
+    const contract = new Contract(contractConfig.address);
+    const [tokenAddress] = args;
+
+    const operation = contract.call('register_canonical_token', nativeToScVal(tokenAddress, { type: 'address' }));
+
+    await broadcast(operation, wallet, chain, 'Canonical Token Registered', options);
+}
+
+async function createPayGasAndCallContractAuth(spenderScVal, payload, gasTokenAddress, gasFeeAmount, chain, wallet) {
```

Comment:
Thought more about it, can you add `caller.require_auth` at the top-level of ITS call to rely on the simulation for now to keep it simpler? Doing all this in the UI is going to be unnecessarily complex